### PR TITLE
Migrate Podcast Episodes to a Content Collection

### PR DESCRIPTION
**Pull request by andygrunwald**
This Pull Request migrates all Podcast Episodes into a Content Collection.

The main changes in short:
* Move files from `/src/pages/podcast/episode/` to `/src/content/podcast/`
* Create schema for each podcast episode (see `src/content/config.ts`)
* Switch data access for an episode from `frontmatter.*` to `data.*`
* Remove the key `layout` from each episode
* Change type `pubDate` from each episode from string to date
* Switch URL generation from `${episode.url}` to `/podcast/episode/${episode.slug}/`
* Enable usage of TypeScript (see `tsconfig.json`)

The optimal result is: Nothing is changing for the look and feel for the end user.

This is one step towards image processing via Astro.

What is missing:
* RSS Feed
* Sitemap testing

Reference:
* [Astro docs: Tutorial ⁠-⁠ Extend with Content Collections](https://docs.astro.build/en/tutorials/add-content-collections/)
* [Astro docs: Content Collections](https://docs.astro.build/en/guides/content-collections/)

Related: #624

### CODE REVIEW

**Review by timhannemann — APPROVED** [review]

File: src/env.d.ts
Comment (woolfg) [inline src/env.d.ts:1]: this file is empty, can we delete it?

**Review by woolfg — APPROVED** [review]
didn't test the scripts, everything else looks good.

tested: shorturls ep95, tracking params, time params, umlaut urls

File: src/env.d.ts
Comment (andygrunwald) [reply src/env.d.ts:1]: Where is this file empty?

It has two lines:
```
/// <reference path="../.astro/types.d.ts" />
/// <reference types="astro/client" />
```

Comment (woolfg) [reply src/env.d.ts:1]: ah ok, that is not a comment? ok, then nevermind

**Pull request merged**